GenericNode/src/main/java/genericnode/GenericNode.java

```diff
@@ -45,6 +56,17 @@ public static void main(String[] args) throws IOException {
                 String key = (args.length > 3) ? args[3] : "";
                 String val = (args.length > 4) ? args[4] : "";
                 // insert code to make RMI client request
+                try {
+                    String name = "Store";
+                    Registry registry = LocateRegistry.getRegistry(addr);
+                    Store stub = (Store) registry.lookup(name);
+                    RMIClient client = new RMIClient(stub);
+                    client.executeTask(cmd, key, val);
```

Comment:
move all this to RMIClient.connect(). Keep the GenericNode.class as clean as possible